enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -76,113 +76,47 @@ public APIKeyAuthenticator() {
-                return requestContext.getHeaders().get(apiKeyName);
-            }
-            if (requestContext.getQueryParameters().containsKey(apiKeyName)) {
-                return requestContext.getQueryParameters().get(apiKeyName);
-            }
-        } else {
-            apiKey = getAPIKey(APIConstants.SWAGGER_API_KEY_AUTH_TYPE_NAME, requestContext, true);
-            if ("".equals(apiKey)) {
-                apiKey = getAPIKey(APIConstants.SWAGGER_API_KEY_AUTH_TYPE_NAME, requestContext, false);
-            }
-        }
-        return apiKey;
-    }
-
-    private static String getAPIKey(String securitySchemeType, RequestContext requestContext, boolean isResourceLevel) {
-        Map<String, List<String>> securitySchemesToApply;
-        if (isResourceLevel) {
-            securitySchemesToApply = requestContext.getMatchedResourcePath().getSecuritySchemas();
-        } else {
-            securitySchemesToApply = requestContext.getMatchedAPI().getApiSecurity();
-        }
-
+    private static String getAPIKeyFromRequest(RequestContext requestContext) {
         Map<String, SecuritySchemaConfig> securitySchemaDefinitions = requestContext.getMatchedAPI().
                 getSecuritySchemeDefinitions();
-
-        for (String securitySchemeName : securitySchemesToApply.keySet()) {
-            SecuritySchemaConfig securitySchemaDefinition = securitySchemaDefinitions.get(securitySchemeName);
-
-            // We only need apiKey of the given type
-            if (securitySchemaDefinition != null && securitySchemeType.equalsIgnoreCase(
-                    securitySchemaDefinition.getType())) {
-
-                // If Defined in openAPI definition (when not enabled at APIM App level),
-                // key must exist in specified location
-                if (APIConstants.SWAGGER_API_KEY_IN_HEADER.equalsIgnoreCase(
-                        securitySchemaDefinition.getIn())) {
-                    if (requestContext.getHeaders().containsKey(securitySchemaDefinition.getName())) {
-                        return requestContext.getHeaders().get(securitySchemaDefinition.getName());
-                    }
-                }
-                if (APIConstants.SWAGGER_API_KEY_IN_QUERY.equalsIgnoreCase(
-                        securitySchemaDefinition.getIn())) {
-                    if (requestContext.getQueryParameters().containsKey(securitySchemaDefinition.getName())) {
-                        return requestContext.getQueryParameters().get(securitySchemaDefinition.getName());
+        // check both resource level and api level security
+        List<Map<String, List<String>>> securityLists = Arrays.asList(requestContext.getMatchedResourcePath()
+                .getSecuritySchemas(), requestContext.getMatchedAPI().getApiSecurity());
+        for (Map<String, List<String>> securityList : securityLists) {
+            for (String securityDefinitionName : securityList.keySet()) {
```

Comment:
Shall we put a comment here. maybe an example to show why it is a for loop here.

Reply:
added with https://github.com/wso2/product-microgateway/pull/2501/commits/bfd4a9a3442475d917b8d577820f91c50a473793